vendor/bat-native-ads/src/bat/ads/internal/search_engine/search_providers.h

```diff
@@ -49,12 +49,22 @@ const std::vector<SearchProviderInfo> _search_providers = {
       // TODO(https://github.com/brave/brave-browser/issues/8487): Brave Ads
       // search providers definition doesn't match all patterns
       "https://google.com",
-      "https://www.google.com/search?q={searchTerms}",
+      "https://www.google.com/search?{searchTerms}",
```

Comment:
don't we need to include the `q` as key here here so `GetValueForKeyInQuery` knows which value to retrieve? Maybe we can change `SearchProviderInfo` to include a `key` property, then we don't need to do the regex bit on the pattern.